### PR TITLE
Added Hyperparameter Tuning to Random Erasing

### PR DESCRIPTION
**Pull request by JoeCx**
# Overview

**Type of Change:** New Feature

**Summary:** Added to the objective of the original training program for hyperparamter tuning to allow for tuning on the random erasing values. This should allow us to see better values for random erasing and will hopefully increase our models accuracy.

## UI/UX Implementation

No changes made.

## Model Updates

This section describes changes that were made to the models used by your application.

### Data Models

No changes made.

### Object-Oriented Models

No changes made.

## End-to-End Testing Instructions

Run hyperparameter tuning in the hyperparameter_simulator.py file to view the best values for random erasing.

### CODE REVIEW

File: src/training_program.py [hunk 1]
```diff
@@ -443,7 +448,7 @@ def objective(self, trial, view, num_epochs=10, k_folds=3):
                 test_loader=val_loader,
                 view=view,
                 lr=lr,
-                optimizer_type=optimizer_type
+                optimizer_type="adam"
```
Comment (tmckeown311) [inline src/training_program.py:451]: Good removing the option for the other optimizer type as it was never found to be more optimal

File: src/training_program.py [hunk 2]
```diff
@@ -403,15 +403,20 @@ def objective(self, trial, view, num_epochs=10, k_folds=3):
...
         batch_size = trial.suggest_categorical("batch_size", [16, 32, 64])
-        optimizer_type = trial.suggest_categorical("optimizer_type", ["adam", "sgd"])
         rotation = trial.suggest_int("rotation", 0, 20)
         brightness = trial.suggest_float("brightness", 0.0, 0.3)
+        erasing_p = trial.suggest_float('erasing_p', 0.0, 0.8)
+        erasing_scale_min = trial.suggest_float('erasing_scale_min', 0.01, 0.1)
+        erasing_scale_max = trial.suggest_float('erasing_scale_max', 0.1, 0.4)
+
+        if erasing_scale_min >= erasing_scale_max:
+            return 0.0  # Invalid trial
```
Comment (tmckeown311) [inline src/training_program.py:413]: Adding erasing as a parameter should help with maximizing our model's individual generalization

**Review by tmckeown311 — APPROVED** [review]
Good changes. Running in the simulator is successful
